### PR TITLE
Always build optimized versions of a couple slow build deps

### PR DESCRIPTION
**Pull request by jgallagher**
TL;DR: This speeds up clean debug builds of nexus from 2m48s to 2m34s (on my machine, YMMV).

On `main` as of a356cc20, two sore thumbs during a clean build are the build scripts of `samael` and `polar-core` (the yellow bars):

main-cargo-timing

I tried using [cargo profiles](https://doc.rust-lang.org/cargo/reference/profiles.html#overrides) to optimize build dependencies, but it was a net loss: it shrank the build script runtimes for these two crates, but added considerably more to many more build dependencies, adding almost a full minute to the total build time in net. AFAICT it's not possible to use a cargo profile to specify the build scripts of specific crates; however, in this case we get a bit lucky in that both of the build scripts we're interested in depend on other crates for their heavy lifting. `samael`'s build script depends on `bindgen`, and `polar-core`'s build script depends on `lalrpop`. By optimizing those two crates, we slow down their compilation but speed up the build scripts considerably; it's a clear net win on my machine:

branch-cargo-timing

(The runtime for `polar-core`'s build script has disappeared due to the filter; its runtime on this branch is 1.3s).

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -64,6 +64,12 @@ resolver = "2"
 [profile.dev]
 panic = "abort"
 
+[profile.dev.package.bindgen]
+opt-level = 3
+
+[profile.dev.package.lalrpop]
+opt-level = 3
+
```
[the remark this answers is not in the document]
Comment (jgallagher) [reply Cargo.toml]: Certainly! Added in e04f177
